src/core/parallel/job_idle.cc

```diff
@@ -149,15 +152,19 @@ SleepTask::SleepTask(Job_Idle* idle_job)
 
 void SleepTask::execute() {
   parent_->remove_running_thread();
-  semaphore_.wait();
+  std::unique_lock<std::mutex> lk(cv_m_);
+  cv_.wait(lk,[&]{return job_ != nullptr;});
   xassert(job_);
   thpool->assign_job_to_current_thread(job_);
 }
 
 
 void SleepTask::wake_up(int nth, ThreadJob* next_job) {
-  job_ = next_job;
-  semaphore_.signal(nth);
```

Comment:
`nth` can now be removed from `wake_up()` parameters, since it is not used anymore